backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/vdsbroker/ConnectStorageServerVDSCommand.java

```diff
@@ -43,8 +43,10 @@ protected void executeVDSCommand() {
 
     @Override
     protected void executeVdsBrokerCommand() {
+        List<String> deviceList = getParameters().getDeviceList();
         _result = getBroker().connectStorageServer(getParameters().getStorageType().getValue(),
-                getParameters().getStoragePoolId().toString(), buildStructFromConnectionListObject());
+                getParameters().getStoragePoolId().toString(), buildStructFromConnectionListObject(),
+                deviceList != null ? deviceList.toArray(new String[deviceList.size()]) : null);
```

Comment:
I see a lot of places using arrays instead of list, but I also see lists
https://github.com/oVirt/ovirt-engine/blob/95107002b8fcbd73131bf915a18d613e432ecf37/backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/jsonrpc/JsonRpcVdsServer.java#L2527

So I think it can work without doing these conversions